tha2/app/ifacialmocap_puppeteer.py

```diff
@@ -392,6 +392,20 @@ def update_result_image_panel(self, event: wx.Event):
         output_index = self.output_index_choice.GetSelection()
         output_image = self.poser.pose(self.torch_source_image, pose, output_index)[0].detach().cpu()
         numpy_image = convert_output_image_from_torch_to_numpy(output_image)

```

Comment:
(See the comment on convert_output_image_from_torch_to_numpy  first.) 

Now that convert_output_image_from_torch_to_numpy will output an uint8 tensor, it will not work with the statements below. Please remove this line and replace it with the if statement that you removed earlier:

        if output_image.shape[0] == 4:
            numpy_image = rgba_to_numpy_image(output_image)
        elif output_image.shape[0] == 1:
            c, h, w = output_image.shape
            alpha_image = torch.cat([output_image.repeat(3, 1, 1) * 2.0 - 1.0, torch.ones(1, h, w)], dim=0)
            numpy_image = rgba_to_numpy_image(alpha_image)
        elif output_image.shape[0] == 2:
            numpy_image = grid_change_to_numpy_image(output_image, num_channels=4)
        else:
            raise RuntimeError("Unsupported # image channels: " + output_image.shape[0])